addons/product/static/src/js/product_pricelist_report.js

```diff
@@ -1,317 +1,242 @@
-odoo.define('product.generate_pricelist', function (require) {
-'use strict';
-
-var AbstractAction = require('web.AbstractAction');
-var core = require('web.core');
-var FieldMany2One = require('web.relational_fields').FieldMany2One;
-var StandaloneFieldManagerMixin = require('web.StandaloneFieldManagerMixin');
-var Widget = require('web.Widget');
-
-var QWeb = core.qweb;
-var _t = core._t;
-
-var QtyTagWidget = Widget.extend({
-    template: 'product.report_pricelist_qty',
-    events: {
-        'click .o_remove_qty': '_onClickRemoveQty',
-    },
-    /**
-     * @override
-     */
-    init: function (parent, defaulQuantities) {
-        this._super.apply(this, arguments);
-        this.quantities = defaulQuantities;
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { sprintf } from "@web/core/utils/strings";
+import { useService } from "@web/core/utils/hooks";
+import { useSetupAction } from "@web/webclient/actions/action_hook";
+import { Layout } from "@web/search/layout";
+
+const { Component, useState, onWillStart, markup, onRendered } = owl;
+
+
+export class GeneratePriceListReport extends Component {
+    setup() {
+        this.orm = useService("orm");
         this.MAX_QTY = 5;
```

Comment:
```suggestion
    setup() {
        this.action = useService("action");
        this.orm = useService("orm");
        this.MAX_QTY = 5;
```

group the services definition (alphabetically) ;).